### PR TITLE
UI: New share button and toolbar reorganize

### LINKED ISSUE #71539: Improve the "Public" tag functionality
### Background / Why we're doing this

The tag for "Public" makes it look like it’s a button. But might be solved by using the badge component, using an outline style. 

### Problems or tasks

The "Public" tag looks like a button, but is not clickable:
image (23)

Need to investigate if this is an actual problem where this tag needs to be a button or just change its style and not look like a button.

### Deliverables

- Figma mockup
```[tasklist]
### Tasks
- [x] UX: research options
- [x] Design Mockups
- [x] Feedback from DS squad
```

### LINKED ISSUE #77298: Implement and test

### PR DESCRIPTION
**Pull request by evictorero**
**What is this feature?**
Changes style of the public dashboard badge, the add and share button and reorganize toolbar buttons.

Before
image

Now
image
image

**Which issue(s) does this PR fix?**:

Fixes #77298
Fixes #71539

**Special notes for your reviewer:**
This is the first step of toolbar redesign, more info [here](https://www.figma.com/file/AjvrAorYpnfSoMpU5kwf39/Adoption-of-Sharing?type=design&node-id=24-346&mode=design&t=uBXo0d8G2tnZcQR9-0)
 
Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by natydej** [comment]
We decided to enable a feature flag to control and test this version V1, and get some feedback from users to see how it will impact the usage.

V1

[Figma](https://www.figma.com/file/AjvrAorYpnfSoMpU5kwf39/Adoption-of-Sharing?type=design&node-id=24-346&mode=design)

cc @thanos-karachalios

**Comment by evictorero** [comment]
Blocked until we implement tracking in the toolbar actions.

**Review by lucychen-grafana — COMMENTED** [review]
Tested and LGTM

**Comment by natellium** [comment]
@evictorero Something I just thought is that we should make sure no docs need to be updated (or that they get updated if needed), we may be using some screenshots that need to be renewed.

**Comment by evictorero** [comment]
> @evictorero Something I just thought is that we should make sure no docs need to be updated (or that they get updated if needed), we may be using some screenshots that need to be renewed.

We will wait a few days to collect analytics data from the old button before merge this one. I don't think we update screenshots in our docs for these small changes do we? In any case, this will change a lot over the next weeks/months maybe we can wait until a more refined version?

**Comment by natellium** [comment]
> We will wait a few days to collect analytics data from the old button before merge this one. I don't think we update screenshots in our docs for these small changes do we? In any case, this will change a lot over the next weeks/months maybe we can wait until a more refined version?

I actually don't know :) @imatwawana any thoughts?

**Comment by imatwawana** [comment]
> > We will wait a few days to collect analytics data from the old button before merge this one. I don't think we update screenshots in our docs for these small changes do we? In any case, this will change a lot over the next weeks/months maybe we can wait until a more refined version?
> 
> I actually don't know :) @imatwawana any thoughts?

We can probably hold off on updating screenshots if there are going to be several changes, but the text should be updated as elements get changed. So the docs where we refer to the "sharing icon" should be updated along with this change.

**Review by juanicabanas — APPROVED** [review]
LGTM! 
As we discussed, I'm a little worried about the button padding and height, which is not the same as in the design.
I agree we shouldn't customize the CSS as it was before, though.

**Review by ivanortegaalba — APPROVED** [review]
Looks great to me! 💯 

Please wait for the review from @Ijin08 if he is not aware of the redesign 🙏

File: public/app/features/dashboard/components/DashNav/DashNav.tsx
Comment (evictorero) [reply public/app/features/dashboard/components/DashNav/DashNav.tsx:313]: I searched for this flag and it is being used in more places, I prefer to leave it here to be removed by a cleanup PR.

**Review by natydej — APPROVED** [review]
LGTM!

**Pull request merged**
